packages/cli/src/cmds/validator/keymanager/decryptKeystores/threadPool.ts

```diff
@@ -0,0 +1,67 @@
+import {spawn, Pool, Worker, ModuleThread, QueuedTask} from "@chainsafe/threads";
+import {DecryptKeystoreArgs, DecryptKeystoreWorkerAPI} from "./types.js";
+import {maxPoolSize} from "./poolSize.js";
+
+/**
+ * Thread pool to decrypt keystores
+ */
+export class DecryptKeystoresThreadPool {
+  private pool: Pool<ModuleThread<DecryptKeystoreWorkerAPI>>;
+  private tasks: QueuedTask<ModuleThread<DecryptKeystoreWorkerAPI>, Uint8Array>[] = [];
+  private terminatePoolHandler: () => void;
+
+  constructor(keystoreCount: number, private readonly signal: AbortSignal) {
+    this.pool = Pool(
+      () =>
+        spawn<DecryptKeystoreWorkerAPI>(new Worker("./worker.js"), {
+          // The number below is big enough to almost disable the timeout
+          // which helps during tests run on unpredictably slow hosts
+          timeout: 5 * 60 * 1000,
+        }),
+      {
+        // Adjust worker pool size based on keystore count
+        size: Math.min(keystoreCount, maxPoolSize),
+        // Decrypt keystores in sequence, increasing concurrency does not improve performance
```

Comment:
Did several test runs with different concurrency but it seems to just overload the threads and throttle the CPU